From 28982bb2d69555233ab4db1de16bb7b4c2f7246d Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 23 Feb 2022 09:12:35 +0200
Subject: [PATCH] [SW-69581] gaudi3: memory: use sram for preboot and fit

The Sram memory of gaudi3 is set to single die mode by default, and
it has been validated already, so we can use it for preboot execution
and images storage.

Change-Id: I07bcabd30f8349cc019259fed24c73da247549ea
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index f2dcabc..4883ebe 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -271,54 +271,6 @@
 	/* TODO: add cpu_dyn_regs asigments */
 }
 
-/*
- * due to the complexity of SRAM initialization, for phase1 of testing
- * spsram memory will be temporary used as storage.
- * first 4k of the spsram are left for the boot info.
- * due to overall limited functionality, at the moment no data overlapping
- * is expected.
- * TODO: after SRAM memory init is implemented, return the memory regions to
- * SRAM.
- */
-
-#define GAUDI3_TMP_MEM		(SP_SRAM_BASE + SZ_4K)
-#define GAUDI3_TMP_MEM_SZ	(HL_SP_SRAM_SIZE - SZ_4K)
-
-static struct mem_desc mem_desc[MEM_NOENT] = {
-	[MEM_GP] = {
-		.base = GAUDI3_TMP_MEM,
-		.size = GAUDI3_TMP_MEM_SZ,
-	},
-	[MEM_LKD_PCI] = {
-		.base = GAUDI3_TMP_MEM,
-		.size = GAUDI3_TMP_MEM_SZ,
-	},
-	[MEM_FWUPD_PCI] = {
-		.base = GAUDI3_TMP_MEM,
-		.size = GAUDI3_TMP_MEM_SZ,
-	},
-	[MEM_I2C] = {
-		.base = GAUDI3_TMP_MEM,
-		.size = GAUDI3_TMP_MEM_SZ,
-	},
-};
-
-int preboot_get_dev_spec_mem_region(enum mem_region region, struct mem_desc *desc)
-{
-	if (region >= MEM_NOENT) {
-		preboot_puts_hex("unknown region = ", region, true);
-		return 1;
-	}
-
-	if (!desc)
-		return 1;
-
-	desc->base = mem_desc[region].base;
-	desc->size = mem_desc[region].size;
-
-	return 0;
-}
-
 bool preboot_is_valid_address_range(u64 addr, int len)
 {
 	if (addr >= SP_SRAM_BASE &&
@@ -337,6 +289,9 @@
 	    addr + len <= ARC1_ICCM_FULL_BASE + ARC1_MEM_SIZE)
 		return true;
 
+	if (addr >= SDRAM_BASE && addr + len <= SDRAM_BASE + SDRAM_SIZE)
+		return true;
+
 	preboot_puts_hex64("ERROR: invalid address range addr = ", addr, true);
 
 	return false;
diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index cdf4f12..a32e00d 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -39,11 +39,7 @@
 #define ARC0_ICCM_BASE			(FW_MEM_REGION_BASE + 0x3000000)
 #define HL_SPI_BASE			(FW_MEM_REGION_BASE + 0x2000000)
 
-/*
- * preboot is running from the upper half of Arc1 ICCM, and loads arcmgmt app
- * to the lower half of Arc1 ICCM
- */
-#define PREBOOT_TEXT_BASE		(ARC1_ICCM_SIZE / 2)
+#define PREBOOT_TEXT_BASE		SDRAM_BASE
 #define PPBOOT_TEXT_BASE		FW_SRAM_BASE
 #define RECOVERY_TEXT_BASE		0x0
 #define RECOVERY_SIZE			SZ_16K
